### PR TITLE
KBV-308 - decrypt JAR and verify EC signature

### PR DESCRIPTION
**Pull request by tal-nagra**
### What changed

The session lambda function now decrypts the incoming JAR 
Support has been added for JWT signature verification using the EC signing algorithm

### Issue tracking
- [KBV-308](https://govukverify.atlassian.net/browse/KBV-308)

### CODE REVIEW

**Review by KaraokeStu — APPROVED** [review]
All looks good to me - it may even fix the issue I've been having, but even if it doesn't, I can use the objectmapper to sort the fix quickly

**Review by idrop — DISMISSED** [review]

File: lambdas/session/src/main/java/uk/gov/di/ipv/cri/address/api/handler/SessionHandler.java [hunk 1]
```diff
@@ -70,7 +70,7 @@ public APIGatewayProxyResponseEvent handleRequest(
 
         } catch (SessionValidationException e) {
 
-            eventProbe.log(INFO, e).counterMetric(EVENT_SESSION_CREATED, 0d);
+            eventProbe.log(ERROR, e).counterMetric(EVENT_SESSION_CREATED, 0d);
```
Comment (idrop) [inline lambdas/session/src/main/java/uk/gov/di/ipv/cri/address/api/handler/SessionHandler.java:73]: I'd say this is still INFO and not an ERROR if we had an issue with the request

Comment (tal-nagra) [reply lambdas/session/src/main/java/uk/gov/di/ipv/cri/address/api/handler/SessionHandler.java:72]: I've never logged an exception with anything other than log level ERROR tbf. I'm not sure if the INFO log level provides the same level of verbosity as the ERROR level

**Review by KaraokeStu — APPROVED** [review]
This all looks good now, but will still need my fixes applying to it (should be simple to merge)

**Pull request merged**